cmd/osbuild-composer/composer.go

```diff
@@ -99,15 +100,17 @@ func (c *Composer) InitWeldr(repoPaths []string, weldrListener net.Listener) err
 		name = "centos-stream-8"
 	}
 
-	repos, err := rpmmd.LoadRepositories(repoPaths, name)
+	rr, err := reporegistry.New(repoPaths)
 	if err != nil {
-		return fmt.Errorf("Error loading repositories for %s: %v", hostDistro.Name(), err)
+		return fmt.Errorf("error loading repositories definitions: %v", err)
 	}
 
+	// TODO: should we check here if repositories for the host distro and arch were loaded?
```

Comment:
So the code previously failed here if the distro repo definitions were not found. Now it will fail later, when a blueprint is depsolved or compose created. Should we check this here and fail immediately?

Reply:
For now I think it should keep the same behavior. I think @teg has mentioned that eventually we should have a fallback distro so that if it doesn't match exactly it will still run, but that's a separate thing to work on.

Reply:
OK, failing right away makes sense from my point of view for now, as it keeps the current behavior.

Reply:
I added the check.